### PR TITLE
Fix mem leak in MegaSyncEventPrivate

### PR DESCRIPTION
**Pull request by cb-mega**

### CODE REVIEW

File: include/megaapi_impl.h [hunk 1]
```diff
@@ -919,20 +919,20 @@ class MegaContactRequestPrivate : public MegaContactRequest
 
 #ifdef ENABLE_SYNC
 
-class MegaSyncEventPrivate: public MegaSyncEvent
+class MegaSyncEventPrivate : public MegaSyncEvent
 {
 public:
+    explicit
     MegaSyncEventPrivate(int type);
```
Comment (mattw-mega) [inline include/megaapi_impl.h]: this is just one declaration; it should be just one line.

File: include/megaapi_impl.h [hunk 2]
```diff
@@ -919,20 +919,20 @@ class MegaContactRequestPrivate : public MegaContactRequest
...
-    virtual int getType() const;
-    virtual const char *getPath() const;
-    virtual MegaHandle getNodeHandle() const;
-    virtual const char *getNewPath() const;
-    virtual const char* getPrevName() const;
-    virtual MegaHandle getPrevParent() const;
+    int getType() const override;
+    const char* getPath() const override;
+    MegaHandle getNodeHandle() const override;
+    const char* getNewPath() const override;
```
Comment (mattw-mega) [inline include/megaapi_impl.h]: No need to switch the space and * around; others are switching them to be the other way in other PRs, and it's all unnecessary - just leave those as they are, keep the PR change to a minimum.

File: src/megaapi_impl.cpp [hunk 3]
```diff
@@ -23066,61 +23056,49 @@ int MegaSyncEventPrivate::getType() const
...
+    return path.get();
 }
 
 MegaHandle MegaSyncEventPrivate::getNodeHandle() const
 {
     return nodeHandle;
 }
 
-const char *MegaSyncEventPrivate::getNewPath() const
+const char* MegaSyncEventPrivate::getNewPath() const
```
Comment (mattw-mega) [inline src/megaapi_impl.cpp]: no need to be switching space and * around

File: include/megaapi_impl.h (hunk 1 above)
Comment (cb-mega) [reply include/megaapi_impl.h]: Is that part of our coding guideline?

File: include/megaapi_impl.h (hunk 2 above)
Comment (cb-mega) [reply include/megaapi_impl.h]: `const char*` is in our coding guideline, `const char *` is not so we should change it if we're making changes anyway.

File: src/megaapi_impl.cpp (hunk 3 above)
Comment (cb-mega) [reply src/megaapi_impl.cpp]: `const char*` is in our coding guideline, `const char *` is not so we should change it if we're making changes anyway.

**Pull request merged**